DXRModules/DeusEx/Classes/DXRFixup.uc

```diff
@@ -797,6 +798,38 @@ static function FixConversationAddNote(Conversation c, string textSnippet)
     }
 }
 
+function SetLampState(#var(prefix)Lamp l, bool turnOn) // could maybe go in DXRLamp if it existed
+{
+    if (turnOn && !l.bOn) {
+        l.bOn = True;
+        l.LightType = LT_Steady;
+        // l.PlaySound(sound'Switch4ClickOn');
+        l.bUnlit = True;
+        l.ScaleGlow = 3.0;
+    } else if (!turnOn && l.bOn) {
+        l.bOn = False;
+        l.LightType = LT_None;
+        // l.PlaySound(sound'Switch4ClickOff');
+        l.bUnlit = False;
+        l.ResetScaleGlow();
+    }
+}
+
+function SetAllLampsState(bool type1, bool type2, bool type3)
+{
+#ifdef vanilla
+    local #var(prefix)Lamp l;
+
```

Comment:
I really do feel like this needs a config option, at least for now.

Reply:
There's certainly no harm in a config option but this really seems like the way it should be. I'd even be inclined to enable it by default in Zero Rando mode.